### ISSUE TITLE
`Restangular.copy` seemingly changes URL port

### ISSUE THREAD

**Issue by rhys-vdw** [mgonto/restangular#748]
Hi there. Perhaps beginner confusion, but seems to be more likely a bug.

Here is some code that demonstrates the problem:

``` coffee
  .controller 'ClientIssuesCtrl', (Restangular, $stateParams) ->
    ctrl =
      issues: []
      isLoading: true
      toggleClosed: (issue) ->
        issue.save()
        issueCopy = Restangular.copy issue
        issueCopy.isClosed = ! issue.isClosed
        issueCopy.save()
```

`issue.save()` hits the correct URI (in this case `http://localhost:5000/users/1/issues/11`), but `issueCopy.save()` gets a 404 as for some reason it tries to `PUT` to `http://localhost:9000/users/1/issues/11`.

Any idea how I can stop this from happening? I'm using Restangular 1.4.0.

**Comment by rhys-vdw** [comment]
@pauldijou Hey, so I found the solution to this, but it's not well documented and also perhaps not the best behaviour...

Basically it's a problem where the Restangular instance doing the clone is copying some data from itself (rather than the instance that created the route object). Hopefully that makes sense. So in my case I created an instance with `Restangular.withConfig` and then used `Restangular.clone()` to copy it. Because I hadn't configured my `Restangular` injectable it did not have a `baseUrl`.

I would suggest that the `clone` method should check which instance created the route and copy the port and `baseUrl` correctly.

**Comment by jdalt** [comment]
@rhys-vdw I just experienced the same issue. If you use the element based `.clone()` method on a Restangular object instance that was created `.withConfig` then you will get the config of the global Restangular configuration. The workaround is to inject `Restangular.withConfig` object and perform a `copy`.

I think this was an oversight in how the `clone` method was implemented (background: https://github.com/mgonto/restangular/issues/299). It seems like the method was created as an alias to `Restangular.copy` which doesn't account for the case of multiple Restangular configurations that can create objects. It is quite surprising when the cloned object doesn't behave like the original. The semantics of clone suggest the new object will behave exactly as the original and it doesn't.
